packages/lisk-transactions/src/transactions/7_out_transfer_transaction.ts

```diff
@@ -0,0 +1,460 @@
+import {
+	Attributes,
+	BaseTransaction,
+	ENTITY_ACCOUNT,
+	ENTITY_TRANSACTION,
+	EntityMap,
+	RequiredState,
+	TransactionResponse,
+} from './base';
+
+const TRANSACTION_DAPP_REGISTER = 5;
+
+export interface OutTransferAsset {
+	readonly outTransfer: {
+		readonly dappId: string;
+		readonly transactionId: string;
+	};
+}
+
+export interface RequiredOutTransferState extends RequiredState {
+	readonly recipient?: Account;
+	readonly dependentState?: {
+		readonly [ENTITY_TRANSACTION]: ReadonlyArray<TransactionJSON>;
+	};
+}
+
+export const outTransferAssetTypeSchema = {
+	type: 'object',
+	required: ['outTransfer'],
+	properties: {
+		outTransfer: {
+			type: 'object',
+			required: ['dappId', 'transactionId'],
+			properties: {
+				dappId: {
+					type: 'string',
+				},
+				transactionId: {
+					type: 'string',
+				},
+			},
+		},
+	},
+};
+
+export const outTransferAssetFormatSchema = {
+	type: 'object',
+	required: ['outTransfer'],
+	properties: {
+		outTransfer: {
+			type: 'object',
+			required: ['dappId', 'transactionId'],
+			properties: {
+				dappId: {
+					type: 'string',
+					format: 'id',
+				},
+				transactionId: {
+					type: 'string',
+					format: 'id',
+				},
+			},
+		},
+	},
+};
+
+export class OutTransferTransaction extends BaseTransaction {
+	public readonly asset: OutTransferAsset;
+	public readonly containsUniqueData = true;
+	public readonly fee: BigNum = new BigNum(OUT_TRANSFER_FEE);
+
+	public constructor(tx: TransactionJSON) {
+		super(tx);
+		const typeValid = validator.validate(outTransferAssetTypeSchema, tx.asset);
+		const errors = validator.errors
+			? validator.errors.map(
+					error =>
+						new TransactionError(
+							`'${error.dataPath}' ${error.message}`,
+							tx.id,
+							error.dataPath,
+						),
+			  )
+			: [];
+		if (!typeValid) {
+			throw new TransactionMultiError('Invalid field types', tx.id, errors);
+		}
+		this.asset = tx.asset as OutTransferAsset;
+	}
+
+	public static fromJSON(tx: TransactionJSON): OutTransferTransaction {
+		const transaction = new OutTransferTransaction(tx);
+		const { errors, status } = transaction.validateSchema();
+
+		if (status === Status.FAIL && errors.length !== 0) {
+			throw new TransactionMultiError(
+				'Failed to validate schema.',
+				tx.id,
+				errors,
+			);
+		}
+
+		return transaction;
+	}
+
+	protected getAssetBytes(): Buffer {
+		const { dappId, transactionId } = this.asset.outTransfer;
+		const outAppIdBuffer = Buffer.from(dappId, 'utf8');
+		const outTransactionIdBuffer = Buffer.from(transactionId, 'utf8');
+
+		return Buffer.concat([outAppIdBuffer, outTransactionIdBuffer]);
+	}
+
+	public assetToJSON(): object {
+		return {
+			...this.asset,
+		};
+	}
+
+	public getRequiredAttributes(): Attributes {
+		const attr = super.getRequiredAttributes();
+		const accounts = {
+			...attr[ENTITY_ACCOUNT],
+			address: [...attr[ENTITY_ACCOUNT].address, this.recipientId],
```

Comment:
This looks like it's incorrect. First, we are destructuring the returned value of `super` and again using the destructured value in the array of address property.

Reply:
since `attr.account.address` is an array, so in order to make it immutable, i think it's needed?